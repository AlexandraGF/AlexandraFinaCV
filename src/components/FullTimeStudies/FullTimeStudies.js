import React, { Component } from 'react';
import './FullTimeStudies.scss';

class FullTimeStudies extends Component {
    constructor(props) {
        super(props);
        this.state = {clicked: false};
    
        this.handleClick = this.handleClick.bind(this);
      }

      handleClick() {
        this.setState(state => ({
          clicked: !state.clicked
        }))
    };

    render() {
        const studiesList = [
            {id: 1, school: 'Romanian-American University, Bucharest', degree: 'Master’s in Strategic Business Management', time: 'October 2011 – June 2013', grade: 'First Class Honours, ranked 3rd out of 41 graduates'},
            {id: 2, school: 'Romanian-American University, Bucharest', degree: 'Bachelor’s in Management', time: 'October 2008 – June 2011', grade: 'First Class Honours, ranked 5th out of 181 graduates'},
            {id: 3, school: 'Gheorghe Lazar National College, Bucharest', degree: 'Diploma - Mathematics and Computer Science (A’ level equivalent)', time: 'September 2004 – June 2008', grade: '86.5%'}
        ];

        return (
        <div>
            <div className="studiesHeading" onClick={this.handleClick}>
                <div className="studiesTitle">Full Time Studies</div>
                <div className="studiesIconImage">{!this.state.clicked ? "+" : "-"}</div>
            </div>
            <div className="studiesBody">{this.state.clicked ? (<div>{studiesList.map(function(item, i){
                return (
                <div className="listItemStudies">
                <div className="listStudiesLine">{item.school}</div>
                <div className="listStudiesLine">{item.degree}</div>
                <div className="listStudiesLine">{item.time}</div>
                <div className="listStudiesLine">Grade: {item.grade}</div>
                </div>
                );
                })}
                </div>
                ) : ""} 
            </div>
        </div>
        );
    }
}

export default FullTimeStudies;
