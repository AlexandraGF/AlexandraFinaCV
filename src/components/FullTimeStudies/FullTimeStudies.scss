@import '../../variables.scss';

.studiesHeading {
    font-family: 'Allura', cursive;
    font-size: 30px;
    margin: 7% 7% 0 11%;
    padding-left: 6%;
    box-shadow: $boxShadow;
    border-radius: $borderRadius;
    display: inline-block;
    width: 80%;
}

.studiesTitle {
    width: 80%;
}

.studiesIconImage {
    width: 8%;
}

.studiesTitle, .studiesIconImage {
    flex-wrap: wrap;
    display: inline-flex;
}

.studiesHeading:hover {
    transform: $hoverSize;
    cursor: $cursorPointer;
}

.studiesBody {
    box-shadow: 5px 10px 12px #a2c3c3;
    margin-top: 7%;
    margin-left: 7%;
    font-family: 'K2D', sans-serif;
}

.listStudiesLine {
    margin-bottom: 2%;
}

.listItemStudies {
    margin: 0 5% 5% 5%;
    padding: 1% 1% 1%;
    box-shadow: $boxShadow;
}
