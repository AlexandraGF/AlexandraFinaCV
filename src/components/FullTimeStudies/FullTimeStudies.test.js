import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

import FullTimeStudies from './FullTimeStudies';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<FullTimeStudies />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders text', () => {
    const wrapper = mount(<FullTimeStudies />);
    expect(wrapper.text()).toContain("Full Time Studies");
    });
    
    it('renders FullTimeStudies Component', () => {
    const tree = renderer.create(
    <FullTimeStudies />
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
    })
