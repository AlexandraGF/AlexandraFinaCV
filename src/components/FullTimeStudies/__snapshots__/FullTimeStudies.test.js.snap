// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App Component 1`] = `
<div>
  <div
    className="studiesHeading"
    onClick={[Function]}
  >
    <div
      className="studiesTitle"
    >
      Full Time Studies
    </div>
    <div
      className="studiesIconImage"
    >
      <img
        src="plus.png"
      />
    </div>
  </div>
  <div
    className="studiesBody"
  >
    
  </div>
</div>
`;

exports[`renders FullTimeStudies Component 1`] = `
<div>
  <div
    className="studiesHeading"
    onClick={[Function]}
  >
    <div
      className="studiesTitle"
    >
      Full Time Studies
    </div>
    <div
      className="studiesIconImage"
    >
      <img
        src="plus.png"
      />
    </div>
  </div>
  <div
    className="studiesBody"
  >
    
  </div>
</div>
`;
